qiita_db/artifact.py

```diff
@@ -269,107 +284,163 @@ def create(cls, filepaths, artifact_type, name=None, prep_template=None,
             raise qdb.exceptions.QiitaDBArtifactCreationError(
                 "at least one filepath is required.")
 
-        # Parents or prep template must be provided, but not both
-        if parents and prep_template:
+        # Check that the combination of parameters is correct
+        counts = Counter([bool(parents or processing_parameters),
```

Comment:
Would casting each of these boolean values to integers make sense instead? That way you would be able to sum all three values and if the result is not equal to one then you would know more than one parameter was specified. Not blocking, just an alternative to using `Counter`.